x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/register.ts

```diff
@@ -35,14 +35,18 @@ export function registerCspmUsageCollector(
       return true;
     },
     fetch: async (collectorFetchContext: CollectorFetchContext) => {
-      const [
-        indicesStats,
-        accountsStats,
-        resourcesStats,
-        rulesStats,
-        installationStats,
-        alertsStats,
-      ] = await Promise.all([
+      const handleResult = <T>(taskName: string, result: PromiseSettledResult<T>) => {
```

Comment:
nit: we are not handling the error here (I don't see logging as handling it) so I'd suggest name that suggests the actual action when the happy flow happens like `extractPromiseValue`